portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -83,9 +80,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("cmnt");
 
+    Document doc = Document.newBuilder().setContent(comment).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    float score = sentiment.getScore();
+    languageService.close();
+
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
+    Translation translation = translate.translate(comment, Translate.TranslateOption.targetLanguage("fr"));
+    String translatedText = translation.getTranslatedText();
+
     Entity taskEntity = new Entity("Comment");
     taskEntity.setProperty("number", commentData.getSize() + 1);
```

Comment:
I know it's not part of this change, but in general you shouldn't use both an in-memory and a permanent datastore. It would be very easy for `commentData` and the datastore to get out of sync.

Consider this scenario:
 * User loads the page (populates `commentData`)
 * Your server is restarted (this can happen *at any time*)
 * User submits a post (`commentData` is empty)

Furthermore, App Engine could scale up your service to have multiple instances of this servelet, and they could have different versions of `commentData`.

Food for thought: how would you display a number with each comment, without storing a local copy of the comments?
Hint: what does the number actually represent? What is the property you want the comment number to have?